main.js

```diff
@@ -160,4 +160,177 @@ const inventory = [
     originalStock: 10,
     sold: 8,
   },
-];
\ No newline at end of file
+];
+
+//Opdracht 1a: Hoeveel exemplaren moeten we nog verkopen? Schrijf een functie die dit berekent.
+
+function toBeSold() {
+  let tellerStock=0;
+  let tellerSold=0
+  for (i=0; i<inventory.length;i++) {
+    tellerStock=tellerStock+inventory[i].originalStock;
+    tellerSold=tellerSold+inventory[i].sold;
+  }
+  return (tellerStock-tellerSold);
+}
+
+//Opdracht 1b: Zorg ervoor dat dit aantal in het rood wordt weergegeven op de pagina
+
+const voorraadmelding=document.createElement("p");
+voorraadmelding.textContent="Er moeten nog " + toBeSold() + " producten worden verkocht.";
+voorraadmelding.setAttribute("id", "stock");
+const pagina=document.getElementById("pagina");
+pagina.appendChild(voorraadmelding);
+
+//Opdracht 2a: Gebruik een array-methode om een array te maken met alle tv-type namen.
+
+const tvtypen = inventory.map((tv)=> {
+  return tv.type;
+})
+
+//Opdracht 2b: Gebruik een array-methode om alle tv's te verzamelen (de hele objecten) die volledig uitverkocht zijn.
+//Omgezet naar functie ivm bonusopdracht
+
+function soldOut() {
+    console.log (inventory.filter((tv)=>{
+        return tv.originalStock===tv.sold;
+    }));
+}
+
+//Opdracht 2c: Gebruik een array-methode om alle tv's te verzamelen (de hele objecten) die over AmbiLight beschikken.
+//Omgezet naar functie ivm bonusopdracht
+
+function hasAmbiLight() {
+    console.log(inventory.filter((tv)=>{
+        return tv.options.ambiLight===true;
```

Comment:
Hier had het voldoende `return tv.options.ambiLight;` voldoende geweest, omdat `ambiLight` van zichzelf al een boolean is.